src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Couchbase.Lite.Internal.Query;
+using Couchbase.Lite.Support;
+using JetBrains.Annotations;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Couchbase.Lite
+{
+    public sealed class Collection : ICollection, IDisposable, IChangeObservable<DatabaseChangedEventArgs>
+    {
+        [NotNull]
+        internal ThreadSafety ThreadSafety { get; } = new ThreadSafety();
```

Comment:
I don't think this will be needed.  Collection can use the database threadsafety.

Reply:
Because of the below spec, 
_The Collection(Collection) function is added and the Database(Database) factory function will be deprecated in DataSource._
method returns DatabaseSource(collection, ThreadSafety) and there is no where to access database threadsafety.. Did I miss anything?

Reply:
What I mean is, you could just expose Database.ThreadSafety here from this property instead of creating a new one.